packages/gatsby/cache-dir/dev-loader.js

```diff
@@ -34,6 +34,12 @@ class DevLoader extends BaseLoader {
   doPrefetch(pagePath) {
     return Promise.resolve(require(`./socketIo`).getPageData(pagePath))
   }
+
+  __internal__doesPageExist(rawPath) {
+    const pagePath = findPath(rawPath)
+    const page = this.pageDb.get(pagePath)
+    return !!page && page.notFound !== true
```

Comment:
@pieh I think you will get true from this function if there is no 404 page. As it checks `undefined !== true`

Reply:
I've done some checks for various variants - not most readable - but check:

![Screenshot 2020-02-12 at 23 49 46](https://user-images.githubusercontent.com/419821/74384825-6b8c4280-4df2-11ea-8ec6-6119ba7afe39.png)

Reply:
Ok, I just checked with dev tools debugger. Thanks

Reply:
last one in the screenshot is I think case you mentioned - it will bail early because `page` will be falsy (being undefined)

Reply:
I can add some unit tests for that

Reply:
> Ok, I just checked with dev tools debugger. Thanks

Just for clarification - is the code ok? Or I need to get back to drawing board? :)

Reply:
@pieh for me code is ok, just want to be sure that you are aware of, page.notFound is undefined in cases when there is no 404.js

Reply:
Ok, I was writing some unit tests and see what you mean - I think I will also need to update some runtime code to properly mark `page.notFound` as `true` if 404.html doesn't exist

Reply:
Tho it is bit confusing because the code "as-is" seemed to be working for me, but maybe I messed things up in manual testing